### PR TITLE
BUG GH23282 calling min on series of NaT returns NaT

### LINKED ISSUE #23282: .min() on a series of NaTs returns nan, while .max() returns NaT
#### Code Sample

```python
s = pd.Series([pd.NaT, pd.NaT])

# Returns nan
s.min()

# Returns NaT
s.max()

```
#### Problem description

Calling `.min()` on an series of `NaT` returns `nan`. I would expect this to return `NaT` to as to keep the type the same. Also, whether it returns `nan` or `NaT`, I would expect it to be consistent with `.max()`. I'm on version `0.22`, not `0.23`, so maybe it was fixed in the latest version, but I didn't see an issue addressing it.

#### Expected Output

I would expect that the `s.min()` call would also return `NaT`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.4.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 45 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.22.0
pytest: None
pip: 10.0.1
setuptools: 40.2.0
Cython: None
numpy: 1.15.1
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.5.0
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 1.0.1
sqlalchemy: 1.2.11
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by JustinZhengBC**
- [X] closes #23282 
- [X] tests passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

For `max`, `NaT` values are filled with the lowest possible value. For `min`, they are filled with the highest possible value. The problem is that only the lowest possible value is recognized as `NaT`. Since `nanops.py` is responsible for assigning the highest value to `NaT` when min is called, it should also be responsible for translating it to `NaT` when appropriate.

### CODE REVIEW

**Review by WillAyd — CHANGES_REQUESTED** [review]
Please be sure to always add tests first and foremost

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -718,6 +718,8 @@ def reduction(values, axis=None, skipna=True, mask=None):
                 result = np.nan
         else:
             result = getattr(values, meth)(axis)
+            if is_integer(result) and result == _int64_max:
```
Comment (sinhrks) [inline pandas/core/nanops.py]: It looks affect to integer dtype, `pd.Series([_int64_max]).min() / max()`?

File: pandas/tests/series/test_datetime_values.py [hunk 2]
```diff
@@ -509,3 +509,8 @@ def test_dt_timetz_accessor(self, tz_naive_fixture):
                            time(22, 14, tzinfo=tz)])
         result = s.dt.timetz
         tm.assert_series_equal(result, expected)
+
+    def test_minmax_nat(self):
```
Comment (sinhrks) [inline pandas/tests/series/test_datetime_values.py]: can u add test for timedelta dtype and DataFrame (#10390)

File: pandas/core/nanops.py (hunk 1 above)
Comment (JustinZhengBC) [reply pandas/core/nanops.py]: Fixed, now it only applies the conversion from `_int64_max` to `NaT` if given an appropriate dtype.

File: pandas/tests/series/test_datetime_values.py (hunk 2 above)
Comment (JustinZhengBC) [reply pandas/tests/series/test_datetime_values.py]: Added more tests, but this PR does not fix #10390

File: pandas/core/nanops.py [hunk 3]
```diff
@@ -718,6 +718,9 @@ def reduction(values, axis=None, skipna=True, mask=None):
                 result = np.nan
         else:
             result = getattr(values, meth)(axis)
+            if (is_integer(result) and is_datetime_or_timedelta_dtype(dtype)
```
Comment (jreback) [inline pandas/core/nanops.py]: this needs handling not here but in _wrap_resulf where a scalar should be turned into NaT if it’s null and of the correct dtype

File: pandas/tests/series/test_datetime_values.py [hunk 4]
```diff
@@ -509,3 +509,19 @@ def test_dt_timetz_accessor(self, tz_naive_fixture):
...
+        pd.Series([pd.NaT, pd.Timedelta('nat')]),
+        pd.Series([pd.Timedelta('nat'), pd.Timedelta('nat')])])
+    def test_minmax_nat_series(self, nat):
+        assert nat.min() is pd.NaT
+        assert nat.max() is pd.NaT
+
+    @pytest.mark.parametrize('nat', [
+        pd.DataFrame([pd.NaT, pd.NaT]),
+        pd.DataFrame([pd.NaT, pd.Timedelta('nat')]),
+        pd.DataFrame([pd.Timedelta('nat'), pd.Timedelta('nat')])])
```
Comment (jreback) [inline pandas/tests/series/test_datetime_values.py]: add the commit number

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
i pushed a commit. have a look.

**Review by TomAugspurger — COMMENTED** [review]
Probably needs a release note.

File: pandas/core/nanops.py [hunk 5]
```diff
@@ -266,16 +266,20 @@ def _view_if_needed(values):
     return values
 
 
-def _wrap_results(result, dtype):
+def _wrap_results(result, dtype, fill_value=None):
     """ wrap our results if needed """
 
     if is_datetime64_dtype(dtype):
         if not isinstance(result, np.ndarray):
+            if result == fill_value:
```
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/core/nanops.py]: so this can't be for an i8 type by definition. but yes we can assert it.

**Review by jreback — CHANGES_REQUESTED** [review]
@JustinZhengBC can you add a whatsnew note & some asserts. ping on green.

File: doc/source/whatsnew/v0.24.0.txt [hunk 6]
```diff
@@ -1020,6 +1020,7 @@ Datetimelike
 - Bug in :func:`to_datetime` with an :class:`Index` argument that would drop the ``name`` from the result (:issue:`21697`)
 - Bug in :class:`PeriodIndex` where adding or subtracting a :class:`timedelta` or :class:`Tick` object produced incorrect results (:issue:`22988`)
 - Bug in :func:`date_range` when decrementing a start date to a past end date by a negative frequency (:issue:`23270`)
+- Bug in :func:`min` which would return ``NaN`` instead of ``NaT`` when called on a series of ``NaT`` (:issue:`23282`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.24.0.txt]: Was the bug in the builtin `min` from the standard library, or `Series.min`? Right now, you're linking to the builtin.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @WillAyd over to you.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @JustinZhengBC !

**Pull request merged**